zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+::
+
+  ZIP: 213
+  Title: Shielded Coinbase
+  Owners: Jack Grigg <jack@z.cash>
+  Status: Draft
+  Category: Consensus
+  Created: 2019-03-30
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST" and "MAY" in this document are to be interpreted as described in
+RFC 2119. [#RFC2119]_
+
+The term "network upgrade" in this document is to be interpreted as described in ZIP 200
+[#zip-0200]_.
+
+The term "Sapling" in this document is to be interpreted as described in ZIP 205
+[#zip-0205]_.
+
+The terms "Founders' Reward", "funding stream", and "active funding stream" in this
+document are to be interpreted as described in ZIP 207 [#zip-0207]_.
+
+Abstract
+========
+
+This proposal defines modifications to the Zcash consensus rules that enable coinbase
+funds to be mined to Sapling addresses. It does not disable the use of transparent
+addresses in coinbase transactions.
+
+
+Motivation
+==========
+
+Zcash inherited the concept of "coinbase transactions" from Bitcoin: special transactions
+inside each block that are allowed to have no inputs. These transactions are created by
+miners during block creation, and collect the block reward and transaction fees into new
+transparent outputs that can then be spent. They are also leveraged in Zcash for the
+Founders' Reward (and potentially for funding streams [#zip-0207]_).
+
+On the path to deprecating and removing Bitcoin-inherited transparent addresses within the
+Zcash network, a required step is to be able to create coinbase transactions that have no
+transparent outputs. However, Zcash was launched with a consensus rule preventing coinbase
+transactions from containing shielded outputs, instead enforcing that coinbase funds could
+not be spent in transactions with transparent outputs. This was partly in order to reduce
+the complexity of the original Zcash modifications to the Bitcoin Core codebase, but also
+because at the time, shielded transactions required significant memory and CPU resources
+to create.
+
+The Sapling network upgrade [#zip-0205]_ deployed architectural changes and performance
+improvements that make shielding funds directly in the coinbase transaction feasible. In
```

Comment:
This may be irrelevant but this explanation doesn't quite make sense to me. Why was it banned if the only problem was feasibility? It seems it always could have been optional if miners wanted to do it. Perhaps an explanation would help about why miners are very sensitive to latency in creating a coinbase transaction

Reply:
The problem wasn't only efficiency: "This was partly in order to reduce the complexity of the original Zcash modifications to the Bitcoin Core codebase [...]".